### PR TITLE
expression: implement vectorized evaluation for builtinYearSig

### PR DESCRIPTION
**Pull request by doggeral**
### What problem does this PR solve? 
implement vectorized evaluation for builtinYearSig

### What is changed and how it works?
go test -v -benchmem -bench=TestVectorizedBuiltinTimeFunc -run=TestVectorizedBuiltinTimeFunc
PASS
ok      github.com/pingcap/tidb/expression      0.564s

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by Reminiscent** [comment]
Thanks! @doggeral Could you please add the benchmark result in the PR description? Besides would you like to join our channel #wg-vec-expr in the tidbcommunity slack workspaces for more discussion?

**Comment by Reminiscent** [comment]
To: https://github.com/pingcap/tidb/issues/12101

File: expression/builtin_time_vec.go [hunk 1]
```diff
@@ -58,6 +58,45 @@ func (b *builtinMonthSig) vectorized() bool {
...
+	if err := b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	ds := buf.Times()
+	for i := 0; i < input.NumRows(); i++ {
+		if result.IsNull(i) {
```
Comment (Reminiscent) [inline expression/builtin_time_vec.go]: Here should be the ``` if buf.IsNull(i) {```.

Comment (SunRunAway) [reply expression/builtin_time_vec.go]: It's both ok after calling `MergeNulls`

**Comment by qw4990** [comment]
@doggeral Please sign our CLA to pass the `license/cla` check.

**Pull request merged**